### PR TITLE
feat: user service implementation

### PR DESCRIPTION
**Pull request by aldy505**
Closes # 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

**Review by yogip07 — APPROVED** [review]

File: user/user_profile/exists.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+func (u *Repository) Exists(ctx context.Context, userId int64) (bool, error) {
+	conn, err := u.db.Acquire(ctx)
+	if err != nil {
+		return false, fmt.Errorf("acquiring connection from pool: %w", err)
+	}
+	defer conn.Release()
+
+	tx, err := conn.BeginTx(ctx, pgx.TxOptions{
+		IsoLevel:   pgx.ReadCommitted,
```
Comment (yogip07) [inline user/user_profile/exists.go:17]: kenapa selalu dikasih isolation level? bukannya read committed itu udah default isolation levelnya? @aldy505

File: user/user_profile/create.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+)
+
+func (u *Repository) Create(ctx context.Context, profile UserProfile) error {
+	conn, err := u.db.Acquire(ctx)
+	if err != nil {
+		return fmt.Errorf("acquiring connection from pool: %w", err)
+	}
+	defer conn.Release()
+
+	tx, err := conn.BeginTx(ctx, pgx.TxOptions{
```
Comment (yogip07) [inline user/user_profile/create.go:19]: ini ga pake retry kalau kena `RepeatableRead`?

File: user/user_profile/exists.go (hunk 1 above)
Comment (aldy505) [reply user/user_profile/exists.go:17]: default itu tanpa isolation level

File: user/user_profile/create.go (hunk 2 above)
Comment (aldy505) [reply user/user_profile/create.go:19]: nggak usah, repeatable read kan bakal blocking

File: user/user_profile/exists.go (hunk 1 above)
Comment (yogip07) [reply user/user_profile/exists.go:17]: https://www.postgresql.org/docs/current/transaction-iso.html#XACT-READ-COMMITTED docs nya bilang gini pak

File: user/user_profile/create.go (hunk 2 above)
Comment (yogip07) [reply user/user_profile/create.go:19]: eh btw kenapa dikasih `RepeatableRead` ya? kasusnya kayak gmn? bukannya ini cuman insert biasa dong, yang mana pasti itu row baru

**Review by WahidinAji — APPROVED** [review]
GBTM

**Pull request merged**